examples/docker.yaml

```diff
@@ -56,6 +52,9 @@ probes:
       exit 1
     fi
   hint: See "/var/log/cloud-init-output.log". in the guest
+hostResolver:
+  hosts:
+    host.docker.internal: host.lima.internal
 portForwards:
 - guestSocket: "/run/user/{{.UID}}/docker.sock"
```

Comment:
Can we split this yaml to another PR and hold it until right before the release of vNext, to avoid seeing “this example does not work” issues?

Reply:
I've restored the original script because I realized it is still required when `hostResolver.enabled` is `false`.

I've added comments to the file that should make it obvious that this feature requires lima 0.8.3 or later.

I've also added tests to the PR.